### PR TITLE
Add Named Pipe support and related tests

### PR DESCRIPTION
**Pull request by matthewmeyer**
- Implemented Named Pipe transport in the agents-hosting-directline-namedpipes package.
- Added serialization and deserialization for headers in named pipes.
- Created tests for header serialization/deserialization, named pipe connection, and transport edge cases.
- Introduced a sample NamedPipeAgent demonstrating pipe-only communication.
- Updated observability constants to include metrics for named pipe operations.
- Added TypeScript configuration for the new package and sample agent.

### CODE REVIEW

File: .azdo/ci-pr.yaml [hunk 1]
```diff
@@ -18,9 +18,16 @@ steps:
...
+# "os": ["win32"] because the named-pipe transport only works on Windows.
+# On this Linux pool `npm ci` would fail with EBADPLATFORM for that workspace,
+# so we pass --force to install it anyway. The package's runtime guard
+# (assertWindowsPlatform) still prevents execution on non-Windows hosts, and
+# its tests are platform-skipped accordingly.
 - task: Npm@1
   inputs:
-    command: ci
+    command: custom
+    customCommand: 'ci --force'
```
Comment (benbrown) [inline .azdo/ci-pr.yaml]: are we at risk of missing other problems in the future because of this?

**Review by benbrown — CHANGES_REQUESTED** [review]

**Review by benbrown — APPROVED** [review]
